src/iidm/util/LinkData.cpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/util/LinkData.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace LinkData {
+
+BranchAdmittanceMatrix calculateBranchAdmittance(double r, double x, double ratio1, double angle1,
+                                                 double ratio2, double angle2, const std::complex<double>& ysh1, const std::complex<double>& ysh2) {
+    std::complex<double> a1 = std::polar(ratio1, angle1);
+    std::complex<double> a2 = std::polar(ratio2, angle2);
+
+    std::complex<double> ytr;
+    if (r == 0.0 && x == 0.0) {
+        ytr = std::complex<double>();
+    } else {
+        ytr = std::complex<double>(1) / std::complex<double>(r, x);
+    }
```

Comment:
I would write this differently: from my point of view, the line 23 is useless because you assign the `ytr` variable with the same value. In java, you need to do that because by default, the value is initialized to `null`. 

In C++, you can simply negate the test in order to initialize `ytr`. What do you think?

Reply:
I fully agree with you, it's useless to call default constructor twice :)